keras2onnx/_builtin.py

```diff
@@ -174,6 +194,74 @@ def convert_tf_pack(scope, operator, container):
                               axis=axis)
 
 
+def _convert_tf_pad(scope, operator, container):
+    oopb = OnnxOperatorBuilder(container, scope)
+    node = operator.raw_operator
+    paddings = np.array(_cal_tensor_value(node.inputs[1])).transpose().flatten()
+    mode = node.get_attr("mode") if hasattr(node, 'mode') else None
+    attrs = {}
+
+    if mode:
+        mode = mode.s.decode("utf-8").lower()
+        attrs['mode'] = mode
+    if mode not in [None, "constant"]:
+        raise ValueError(mode + " pad mode is not supported")
+
+    origin_dtype = _to_onnx_type(node.outputs[0].dtype)
+    if origin_dtype not in [onnx_pb.TensorProto.FLOAT16, onnx_pb.TensorProto.FLOAT,
+                            onnx_pb.TensorProto.DOUBLE]:
+        cast_op = oopb.apply_cast(operator.input_full_names[0],
+                                  to=onnx_proto.TensorProto.FLOAT,
+                                  name=operator.full_name + '_cast')
+    else:
+        cast_op = operator.input_full_names[0]
+
+    if operator.target_opset < 11:
+        attrs['pads'] = paddings
+        if mode in [None, "constant"] and len(node.inputs) == 3:
+            const_val = _cal_tensor_value(node.inputs[2]).tolist()
+            attrs['value'] = const_val
+        pad_node =  oopb.add_node("Pad",
+                                  cast_op,
+                                  name=operator.full_name + 'pad',
+                                  op_version=2, **attrs)
```

Comment:
Do we need to account for pad op_version=1 here?

Reply:
I checked tf2onnx code, there is no difference between op_version=2 and op_version=1, we can do op_version=2 for now.